### PR TITLE
Warn when using a minified copy of dev React

### LINKED ISSUE #4842: Warn when using dev build but minified
Do we want to do this now? @sebmarkbage mentioned this but I forgot about it until this morning.

### PR DESCRIPTION
**Pull request by spicyj**
Fixes #4842.

### CODE REVIEW

**Comment by syranide** [comment]
Idea; since you actually enable name mangling and optimizations separately, what about checking `(function() { 0; }).toString().indexOf('0') !== -1`. In the end it doesn't *really* matter, just thought I'd bring it up.

File: src/renderers/dom/ReactDOM.js [hunk 1]
```diff
@@ -74,6 +74,15 @@ if (__DEV__) {
       }
     }
 
+    function testFn() {}
+    warning(
+      (testFn.name || testFn.toString()).indexOf('testFn') !== -1,
```
Comment (gaearon) [inline src/renderers/dom/ReactDOM.js]: Oh, that's smart.

**Comment by jimfb** [comment]
@syranide Haha, interesting.  In theory, a good optimizer would replace `(function() { 0+0; }).toString().indexOf('+') !== -1` with `true`.  But I suspect that's an academic discussion, since I can't think of how it adds much value on top of detecting minification.

**Comment by syranide** [comment]
@jimfb That applies to the original code as well. It would also be an invalid optimization as `toString` could have been overloaded elsewhere.

**Comment by jimfb** [comment]
:+1:

**Comment by lijunle** [comment]
Hi, the short URL https://fb.me/react-minification does not redirect to something meaningful now. There is not download page now.

**Comment by spicyj** [comment]
I updated the URL to point somewhere more relevant.

**Pull request merged**